toolchains/java_toolchain_alias.bzl

```diff
@@ -26,6 +26,8 @@ def _java_runtime_alias(ctx):
         platform_common.TemplateVariableInfo({
             "JAVA": str(toolchain.java_executable_exec_path),
             "JAVABASE": str(toolchain.java_home),
+            "JAVA_RUNFILES": str(toolchain.java_executable_runfiles_path),
```

Comment:
I think we should instead be adding these to https://github.com/bazelbuild/rules_java/blob/af504cff01d441676f2d3e3b64c92d010f85d835/java/common/rules/java_runtime.bzl#L152 and then just forward that provider like we do below in this file: https://github.com/bazelbuild/rules_java/blob/af504cff01d441676f2d3e3b64c92d010f85d835/toolchains/java_toolchain_alias.bzl#L46

Reply:
In order to make this change, I think I would have to make `java_runtime_alias` and `java_host_runtime_alias` identical by adding a dependency on `//toolchains:current_java_runtime` to `java_runtime_alias`. I can do this, but wanted to make sure it's your intent first.

Reply:
My bad, please ignore the bit about just forwarding the provider. But I do think we need to make the same change also in [rules_java/java/common/rules/java_runtime.bzl](https://github.com/bazelbuild/rules_java/blob/af504cff01d441676f2d3e3b64c92d010f85d835/java/common/rules/java_runtime.bzl#L152)

Reply:
I have made the requested changes, however the test I added is failing due to `toolchain.java_executable_runfiles_path` appearing not to be a valid path. It's returning a path like:
```
/private/var/tmp/_bazel_thirtyseven/43339e4e51eaab2f90ecf6c85b4023b4/sandbox/darwin-sandbox/49/execroot/_main/../rules_java++toolchains+local_jdk/bin/java
```
which does not exist when running the test.

Reply:
I have removed `JAVA_RUNFILES` from this PR for now.

Reply:
Realized I could do something like:
```
"JAVA_RUNFILES": toolchain.java_home_runfiles_path + "/bin/java",
```
it's a little ugly but works. Thoughts?

Reply:
Does that work? I would think the issue is that `JAVA_RUNFILES` is a special env var[^1] so if we'll want to use a different name.

[^1]: https://github.com/bazelbuild/bazel/blob/d796714abba0a5531edf795c53a3985300577303/src/main/java/com/google/devtools/build/lib/exec/StandaloneTestStrategy.java#L82

Reply:
Not sure what a better name would be, I'm ok with leaving it out letting the user locate bin/java.